src/features/userSlice.ts

```diff
@@ -0,0 +1,32 @@
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { User } from '../types/User';
+import { getUsers } from '../api/users';
+
+export type UserState = {
+  users: User[];
+};
+
+const initialState: UserState = {
+  users: [],
+};
+
+export const fetchUsers = createAsyncThunk('users/fetch', getUsers);
+
+const usersSlice = createSlice({
+  name: 'users',
+  initialState,
+  reducers: {},
+  extraReducers: builder => {
+    builder.addCase(
+      fetchUsers.fulfilled,
+      (state, action: PayloadAction<User[]>) => {
+        return {
+          ...state,
+          users: action.payload,
+        };
+      },
+    );
+  },
```

Comment:
Consider adding cases for `fetchUsers.pending` and `fetchUsers.rejected` to manage loading and error states. This will help in providing feedback to the user during data fetching operations.